src/sql-parser/src/parser.rs

```diff
@@ -4399,9 +4421,44 @@ impl<'a> Parser<'a> {
         })
     }
 
+    fn parse_option_map<T, F>(
+        &mut self,
+        mut f: F,
+    ) -> Result<Option<BTreeMap<String, T>>, ParserError>
```

Comment:
I'm a little concerned about the differences between this and `parse_map`:
- only strings as keys
- returns a btree instead of a vec of pairs

Only supporting string keys seems maybe ok because this needs to interact with `Value` which isn't an `Expr`. But then returning a `Vec<(String, T)>` would be needless if some other step is going to turn it into a btree anyway. Ok maybe this is fine then?

Reply:
> Ok maybe this is fine then?

Yeah, seems fine to me!

Also, there's no reason we'd ever need to call this with a value type of anything besides `WithOptionValue<T>`, so I'd actually prefer if we dropped the `mut f: F` argument and had this return a `Result<Option<BTreeMap<String, WithOptionValue<Raw>>>>`.